js-apps/meep-frontend/src/js/containers/meep-container.js

```diff
@@ -588,25 +590,53 @@ class MeepContainer extends Component {
    * @param {module:model/GeoDataAssetList} data The data returned by the service call.
    * @param {String} response The complete HTTP response.
    */
-  getComputeAssetDataCb(error, data) {
+  getAssetDataCb(error, data) {
+
     if (error !== null) {
       return;
     }
 
-    // Update Compute list
-    this.props.execChangeMapComputeList(data.geoDataAssets ? _.sortBy(data.geoDataAssets, ['assetName']) : []);
+    var ueList = [];
+    var poaList = [];
+    var computeList = [];
+
+    // Extract assets by type
+
+    if (data.geoDataAssets) {
+      _.forEach(data.geoDataAssets, asset => {
+        switch (asset.assetType) {
+        case 'UE':
+          ueList.push(asset);
+          break;
+        case 'POA':
+          poaList.push(asset);
+          break;
+        case 'COMPUTE':
+          computeList.push(asset);
+          break;
+        default:
+          break;
+        }
+      });
+    }
+    
+    // Update asset map
+    var assetMap = {
+      ueList: _.sortBy(ueList, ['assetName']),
+      poaList: _.sortBy(poaList, ['assetName']),
+      computeList: _.sortBy(computeList, ['assetName'])
+    };
+
+    //Update UE LIST, COMPUTE LIST, POA list
+    this.props.execChangeMapUeList(assetMap.ueList? _.sortBy(assetMap.ueList, ['assetName']) : []);
+    this.props.execChangeMapPoaList(assetMap.poaList? _.sortBy(assetMap.poaList, ['assetName']) : []);
+    this.props.execChangeMapComputeList(assetMap.computeList? _.sortBy(assetMap.computeList, ['assetName']) : []);
```

Comment:
Replace this with a single call to execChangeMap() which is already available in map reducer.